apple/Tests/TestCases/RadixConnect/P2PLinks/RadixConnectPasswordTests.swift

```diff
@@ -0,0 +1,19 @@
+import CustomDump
+import Foundation
+import Sargon
+import SargonUniFFI
+import XCTest
+
+final class RadixConnectPasswordTests: Test<RadixConnectPassword> {
+    func test_codable() throws {
+		let raw = "\"deaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddead\"".data(using: .utf8)!
+		
+		// test decoding
+		let sut = try JSONDecoder().decode(SUT.self, from: raw)
+		XCTAssertEqual(sut, SUT.sample)
+		
+		// test encoding
+		let encoded = try JSONEncoder().encode(sut)
+		try XCTAssertEqual(JSONDecoder().decode(SUT.self, from: encoded), sut)
+	}
```

Comment:
Apart from those JSON tests which tests against known/expected values, you can also use the:

Perhaps add this function in to bottom of `Test.swift` below `func doTestCodableRoundtrip` 

```swift
	func eachSampleCodableRoundtripTest(
		_ test: (SUT) throws -> Void
	) rethrows {
		try eachSample { try doTestCodableRoundtrip($0) }
	}
```

and you can then add a tiny test to this file `RadixConnectPasswordTests.swift` (and all other test files doing JSON test) ? which is:

```swift
/// Cyon: We might be able remove this function once we have converted to `swift-testing` which has much more 
/// powerful discovery than XCTest, and maybe `eachSampleCodableRoundtripTest` will be picked up as
/// a test directly.
func testJSONRoundtripAllSamples() throws {
    try eachSampleCodableRoundtripTest()
}
```

In general I forgot about the `eachSample` function which we wanna try use as much as possible (ping @matiasbzurovski and @GhenadieVP ).

If you find tests like this:

```swift
	func test_phrase_roundtrip() throws {
		func doTest(_ sut: SUT) throws {
			let string = sut.description
			try XCTAssertEqual(SUT(phrase: string), sut)
			try XCTAssertEqual(SUT(phrase: string, language: .english), sut)
			try XCTAssertEqual(SUT(phrase: string, language: nil), sut) // ok to skip language
		}
		try SUT.sampleValues.forEach(doTest)
	}
```

That SHOULD be converted to:

```swift
	func test_phrase_roundtrip() throws {
		try eachSample { sut in
			let string = sut.description
			try XCTAssertEqual(SUT(phrase: string), sut)
			try XCTAssertEqual(SUT(phrase: string, language: .english), sut)
			try XCTAssertEqual(SUT(phrase: string, language: nil), sut) // ok to skip language
		}
	}
```

And if you are working with a type which is `Codable` (`SargonStringCodable` or `SargonObjectCodable`) we should add add a:

```swift
/// Cyon: We might be able remove this function once we have converted to `swift-testing` which has much more 
/// powerful discovery than XCTest, and maybe `eachSampleCodableRoundtripTest` will be picked up as
/// a test directly.
func testJSONRoundtripAllSamples() throws {
    try eachSampleCodableRoundtripTest()
}
```

Reply:
I've done changes for all files in [this](https://github.com/radixdlt/sargon/pull/123/commits/28d2b1842e7f02d778e1c03e466210ec89cbb8a0) commit

Reply:
awesome!